src/wallet/wallet.cpp

```diff
@@ -381,12 +376,10 @@ std::shared_ptr<CWallet> CreateWallet(WalletContext& context, const std::string&
     uint64_t wallet_creation_flags = options.create_flags;
     const SecureString& passphrase = options.create_passphrase;
 
+    // Only descriptor wallets can be created
+    Assert(wallet_creation_flags & WALLET_FLAG_DESCRIPTORS);
+
     if (wallet_creation_flags & WALLET_FLAG_DESCRIPTORS) options.require_format = DatabaseFormat::SQLITE;
```

Comment:
In 5111176aec91396320c7a337c0613389c1a5f4c0 "wallet, refactor: Remove Legacy warnings and errors"

This `if` is not needed anymore with the above `Assert`.

Reply:
maybe neither do this one:

https://github.com/bitcoin/bitcoin/blob/39090518f62cca5432413d1669a85924ba721278/src/wallet/wallet.cpp#L448

(just to confirm and do both in 1 go)

Reply:
Yes